src/interceptors/Socket/MockSocket.ts

```diff
@@ -34,7 +34,7 @@ export class MockSocket extends net.Socket {
   public write(...args: Array<unknown>): boolean {
     const [chunk, encoding, callback] = normalizeWriteArgs(args as WriteArgs)
     this.options.write(chunk, encoding, callback)
-    return false
+    return true
```

Comment:
How does this affect the fix?

I'm hesitant to return `true` because it means that the chunk has been sent. However, this method will be called for request bodies that we forward to the interceptor first. Semantically, we are buffering the request chunks so returning `false` looks more appropriate.

Reply:
We can emit drain instead in `onRequestBody/End` but this would be unnatural as well. 
The more "right" thing to do is to return true/false according to the `this.requestStream.push(chunk)` in the `onRequestBody`

Reply:
My concern is that this is public facing `.write()` while `this.requestStream` is an internal stream we use to feed to the HTTP parser. Returning `true` will _notify the consumer_ (the user/test) that the chunk has been written while it hasn't been. It really gets buffered in the most direct sense of the word. I don't think returning `true` is correct here.

Reply:
We need either return `true` or emit `drain` event

Reply:
Got it. Okay, so let's return `true` since it's the path of least resistance. If it ever causes issues, we iterate. Thanks for pointing this out.